server/cp/ingest/parser/ap.py

```diff
@@ -615,6 +620,38 @@ def _parse_exif(self, data, item):
         if metadata.get('Keywords'):
             item.setdefault('extra', {})[cp.XMP_KEYWORDS] = ', '.join(metadata['Keywords'].split(';'))
 
+    def _find_place(self, data, type_name):
+        return next((place for place in data['item']['place'] if place['locationtype']['name'] == type_name), None)
+
+    def _parse_place(self, data, item):
+        city = self._find_place(data, 'City')
+        state = self._find_place(data, 'State')
+        country = self._find_place(data, 'Nation')
+        region = self._find_place(data, 'Continent')
+        if city:
+            place = {
+                'name': city['name'],
+                'qcode': city['name'],
+            }
+
+            if state:
+                place['state'] = state['name']
+
+            if country:
+                place['country'] = country['name']
+
+            if region:
+                place['world_region'] = region['name']
+
+            geo = city.get('geometry_geojson')
+            if geo and geo['type'] == 'Point':
+                place['location'] = {
+                    'lat': geo['coordinates'][1],
+                    'lon': geo['coordinates'][0],
+                }
```

Comment:
We can wrap this part and catch KeyError just in case.
UPD. We can even wrap an entire `_parse_place` method in place where it's called.